tests/api/test_health.py

```
@@ -70,3 +129,30 @@ def test_health_check_timeout(mock_client_class, client):
     data = response.json()
     assert data["status"] == "degraded"
     assert data["dependencies"]["database"] == "down"
+
+@patch("app.main.httpx.AsyncClient", autospec=True)
+@patch("app.main.get_global_client", return_value=None)
+def test_health_check_fallback_client_used(mock_get_global_client, mock_async_client_cls, client):
```

Ruff (0.15.2)

[warning] Unused function argument: `mock_get_global_client`

(ARG001)